service/handlers_test.go

```diff
@@ -643,6 +643,45 @@ func TestVersionHandler(t *testing.T) {
 
 }
 
+func TestHealthHandlerHealthy(t *testing.T) {
+
+	config := config.Settings{KeyStoreType: "filesystem", KeyStorePath: "../keystore", JwtSecret: "SomeTestSecretValue"}
+	datastore.Environ = &datastore.Env{DB: &datastore.MockDB{}, Config: config}
+	response, _ := sendRequestHealth(t, "GET", "/v1/health", nil)
+
+	if response.Code != 200 {
+		t.Errorf("Response to healthy data store should be 200 but was %v", response.Code)
+	}
+	result := HealthResponse{}
+	err := json.NewDecoder(response.Body).Decode(&result)
+	if err != nil {
+		t.Errorf("Error decoding the health response: %v", err)
+	}
+	if result.Database != "healthy" {
+		t.Errorf("Health endpoint should have returned healthy, instead got %s", result.Database)
+	}
+
+}
+
+func TestHealthHandlerUnhealthy(t *testing.T) {
+
+	config := config.Settings{KeyStoreType: "filesystem", KeyStorePath: "../keystore", JwtSecret: "SomeTestSecretValue"}
+	datastore.Environ = &datastore.Env{DB: &datastore.ErrorMockDB{}, Config: config}
+	response, _ := sendRequestHealth(t, "GET", "/v1/health", nil)
+
+	if response.Code != 500 {
```

Comment:
This test is failing as the method always returns an HTTP 200 response. You'll need to set the response in the HealthCheck method. Also, we prefer HTTP 400 errors for understood application errors - that way we know that if we get an error 500 we know that something unexpected has happened.

So, for example, the health check may actually fail when the service is first deployed as the postgresql unit takes longer to start up and initialize than the SV units.